adapter/internal/discovery/xds/server.go

```diff
@@ -598,6 +598,9 @@ func DeleteAPIWithAPIMEvent(uuid, name, version string, environments []string, o
 			for _, environment := range environments {
 				// delete environment if exists
 				delete(apiUUIDToGatewayToVhosts[uuid], environment)
+				if revisionUUID != "" {
+					notifier.SendRevisionUndeploy(uuid, revisionUUID, environment)
```

Comment:
Let's call this only if Global Adapter is enabled.

Reply:
Fixed via : https://github.com/wso2/product-microgateway/pull/2503/commits/cf1de67208492e8ac71fe95e32a52cf036ab3ee7